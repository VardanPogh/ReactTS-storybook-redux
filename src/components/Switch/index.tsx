import React from 'react'
import { makeStyles, createStyles } from '@material-ui/core/styles'
import Switch from '@material-ui/core/Switch'
import FormControlLabel from '@material-ui/core/FormControlLabel'
import FormControl from '@material-ui/core/FormControl'

const useStyles = makeStyles(() =>
  createStyles({
    switchBase: {
      '&checked': {
        color: '#014bad !important',
        '& + track': {
          backgroundColor: '#014bad !important',
        },
      },
      '&focusVisible thumb': {
        color: '#014bad !important',
      },
    },
    track: {
      backgroundColor: '#014bad !important',
    },
    checked: { color: '#014bad !important' },
  })
)

interface Props {
  labelPlacement: 'bottom' | 'end' | 'start' | 'top'
  label: string
  value: string
  checked?: boolean
  onChange(): void
}

const SwitchStyled: React.FC<Props> = ({
  labelPlacement,
  label,
  value,
  checked,
  onChange,
}) => {
  const classes = useStyles()
  return (
    <FormControl component="fieldset">
      <FormControlLabel
        value={value}
        control={
          <Switch
            classes={{
              switchBase: classes.switchBase,
              track: classes.track,
              checked: classes.checked,
            }}
          />
        }
        checked={checked}
        onChange={onChange}
        label={label}
        labelPlacement={labelPlacement}
      />
    </FormControl>
  )
}

export default SwitchStyled
