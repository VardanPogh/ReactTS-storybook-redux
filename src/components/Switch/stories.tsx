import React, { useState } from 'react'
import { withKnobs, text, select } from '@storybook/addon-knobs'

import Switch from '.'

const labelPlacement = {
  Bottom: 'bottom',
  End: 'end',
  Start: 'start',
  Top: 'top',
}

export default {
  title: 'Switch',
  component: Switch,
  decorators: [withKnobs],
}

export const Default: React.FC = () => {
  const [checked, setChecked] = useState(false)
  const toggleChecked = (): void => {
    setChecked((prev) => !prev)
  }
  return (
    <Switch
      // eslint-disable-next-line
      // @ts-ignore
      labelPlacement={select(
        'labelPlacement',
        labelPlacement,
        labelPlacement.Bottom
      )}
      label={text('Label', 'Sample Label')}
      value={text('Value', 'Sample Value')}
      checked={checked}
      onChange={toggleChecked}
    />
  )
}
