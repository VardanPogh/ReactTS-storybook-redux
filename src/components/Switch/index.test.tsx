import React from 'react'
import { render } from '@testing-library/react'

import Switch from '.'

describe('Switch Test', () => {
  test('Load Switch', () => {
    const myMock = jest.fn()
    const { container } = render(
      <Switch
        labelPlacement="bottom"
        label="Label"
        value="value"
        checked
        onChange={myMock}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
