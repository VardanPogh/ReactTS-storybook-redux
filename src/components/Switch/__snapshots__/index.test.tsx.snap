// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Test Load Switch 1`] = `
<div>
  <fieldset
    class="MuiFormControl-root"
  >
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementBottom"
    >
      <span
        class="MuiSwitch-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiSwitch-switchBase makeStyles-switchBase-1 MuiSwitch-colorSecondary PrivateSwitchBase-checked-5 Mui-checked makeStyles-checked-3"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input-7 MuiSwitch-input"
              type="checkbox"
              value="value"
            />
            <span
              class="MuiSwitch-thumb"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track makeStyles-track-2"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Label
      </span>
    </label>
  </fieldset>
</div>
`;
